app/src/main/java/com/android/unio/ui/event/EventCreation.kt

```diff
@@ -188,7 +220,48 @@ fun EventCreationScreen(
 
           Button(
               modifier = Modifier.testTag(EventCreationTestTags.SAVE_BUTTON),
-              onClick = { navigationAction.goBack() }) {
+              enabled =
+                  name.isNotEmpty() &&
+                      shortDescription.isNotEmpty() &&
+                      longDescription.isNotEmpty() &&
+                      startTimestamp != null &&
+                      endTimestamp != null &&
+                      startTimestamp!! < endTimestamp!! &&
+                      eventBannerUri.value != Uri.EMPTY,
+              onClick = {
+                val inputStream = context.contentResolver.openInputStream(eventBannerUri.value)!!
+                eventViewModel.addEvent(
+                    inputStream,
+                    Event(
+                        uid = "", // This gets overwritten by eventViewModel.addEvent
```

Comment:
It would probably make more sense to create a `getNewUID` function in the `EventViewModel` that calls `repository.getNewUid()`, instead of letting the view model handle this implicitly, simply because it brings less confusion.

Reply:
I'm not sure the UI should interact directly with the repository? But let's discuss this tomorrow